### PR TITLE
Ignore existing event when inserting to fix primary key violation

### LINKED ISSUE #444: Make orderbook maintenance HPA compatible
At the moment, we can run into a primary key violation if two API pods are running at the same time (cf. https://gnosisinc.slack.com/archives/CQZBU9T5J/p1617221794003700).

In the [logs](https://logs.gnosisdev.com/goto/7155413645283910869a632983c2045e) we can see that two pods were operating around the time the error happened. This can happen in case we restart pods (rollin update making sure that one pod has to be fully ready before the existing one is shut down) or more likely when we use high availability auto scaling and create multiple instances of our pod (at the moment this happens on xDAI when our trading bot runs, because it's invoking a lot of price estimates).

The issue seems to be coming from the fact that we first delete all events in one transaction and then insert new events in a separate transaction:

https://github.com/gnosis/gp-v2-services/blob/afec9d968d5e0d824122f7509d42b7072c2e63fe/orderbook/src/event_updater.rs#L72-L83

This allows for race conditions in case there are two instances inserting on the same db.

One way to mitigate would be to bundle the updates in a transaction or add some other way for pods to communicate with one another to signal who is taking care of maintenance.

### PR DESCRIPTION
**Pull request by vkgnosis**
Fixes #444 

### Test Plan
new unit test

### CODE REVIEW

**Review by bh2smith — APPROVED** [review]
Seems like a cheeky fix but should work without a doubt. Did we ever determine where the conflicts were coming from?

**Review by nlordell — APPROVED** [review]
Code looks fine. Same question as :point_up:.

**Comment by vkgnosis** [comment]
See the issue that I linked. Two HPA pods run at same time.

File: orderbook/src/database/events.rs [hunk 1]
```diff
@@ -138,7 +138,8 @@ async fn insert_invalidation(
     event: &Invalidation,
 ) -> Result<(), sqlx::Error> {
     const QUERY: &str =
-        "INSERT INTO invalidations (block_number, log_index, order_uid) VALUES ($1, $2, $3);";
+        "INSERT INTO invalidations (block_number, log_index, order_uid) VALUES ($1, $2, $3) \
```
Comment (fleupold) [inline orderbook/src/database/events.rs:141]: Can we add a comment on why we do the on conflict do nothing here and below (maybe just a link to the issue).

**Review by fleupold — APPROVED** [review]
Alternative would be to acquire a lock early in the transaction (before doing the delete). This might be slightly better to avoid more complex race conditions (e.g. non repeatable reads, etc), but for now this seems like a simple and good enough fix.

File: orderbook/src/database/events.rs (hunk 1 above)
Comment (vkgnosis) [reply orderbook/src/database/events.rs:144]: added

**Pull request merged**